tools/ci_build/build.py

```diff
@@ -892,6 +902,19 @@ def generate_build_tree(cmake_path, source_dir, build_dir, cuda_home, cudnn_home
                 "-DCMAKE_CXX_COMPILER=" + compilers[1]
             ]
 
+    if args.build_wasm:
+        emsdk_dir = os.path.join(cmake_dir, "external", "emsdk")
+        emscripten_cmake_toolchain_file = os.path.join(emsdk_dir, "upstream", "emscripten", "cmake", "Modules",
+                                                       "Platform", "Emscripten.cmake")
+        cmake_args += [
+            "-DCMAKE_TOOLCHAIN_FILE=" + emscripten_cmake_toolchain_file,
+            "-Donnxruntime_ENABLE_NSYNC=OFF"
+        ]
+        if not args.test:
+            cmake_args += [
+                "-Donnxruntime_BUILD_UNIT_TESTS=OFF",
+            ]
+
```

Comment:
onnxruntime_BUILD_UNIT_TESTS doesn't seem like a WASM specific value so not clear why we set it inside 'if args.build_wasm'. Possibly a comment explaining why would be helpful.

Reply:
modified this to checking the flag `--disable_wasm_exception_catching`. also explained in comments.